R/dbpath.R

```diff
@@ -40,9 +40,14 @@ dbpath <- function(url) {
 
     if (length(tokens) > 1) {
       parsed_url[["params"]] <- .rfc_1738_parse_query(tokens[2])
+      parsed_url[["params"]] <- lapply(parsed_url[["params"]], utils::URLdecode)
     }
   }
 
+  if (nzchar(parsed_url[["password"]])) {
+    parsed_url[["password"]] <- utils::URLdecode(parsed_url[["password"]])
```

Comment:
Ah, I'm just now noticing that Ah, I'm noticing that username is also [encoded in sqlalchemy](https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/engine/url.py#L882).. Do you mind adding that piece also?

Reply:
ah yeah that makes sense! don't mind at all 😄